### PR TITLE
citra: add build options

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Motivation for this change
Why not?

###### Things done

- I'm not sure if all of these options should be true by default. Should I respect Citra's CMakeLists.txt defaults or the more stuff is true the better?
- That assert maybe is overkill, if one simply has both options true the only effect is that FDK won't be enabled.
- I changed one option name. It's technically a breaking change, but I doubt that anyone overrode that option.
- I didn't put any option for `ENABLE_QT_TRANSLATION`.
- `-DCITRA_USE_BUNDLED_FFMPEG=OFF` has been removed because it's already off if you are not building on Windows.
- `alsa-lib` was doing nothing here.
- Is `libusb` being devendored if it's just being put in buildInputs?

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/citra/default.nix [hunk 1]
```diff
@@ -33,34 +34,39 @@ stdenv.mkDerivation {
...
+  ++ lib.optionals (enableFfmpegAudioDecoder || enableFfmpegVideoDumper) [ ffmpeg-full ]
+  ++ lib.optionals useDiscordRichPresence [ rapidjson ]
+  ++ lib.optionals enableFdk [ fdk_aac ];
 
   cmakeFlags = [
     "-DUSE_SYSTEM_BOOST=ON"
-    "-DUSE_DISCORD_PRESENCE=ON"
-  ] ++ lib.optionals aacHleDecoding [
-    "-DENABLE_FFMPEG_AUDIO_DECODER=ON"
-    "-DCITRA_USE_BUNDLED_FFMPEG=OFF"
```
Comment (IvarWithoutBones) [inline pkgs/misc/emulators/citra/default.nix:51]: Why was this line removed?

File: pkgs/misc/emulators/citra/default.nix [hunk 2]
```diff
@@ -33,34 +34,39 @@ stdenv.mkDerivation {
...
 
   postPatch = ''
-    # we already know the submodules are present
+    # We already know the submodules are present
     substituteInPlace CMakeLists.txt \
       --replace "check_submodules_present()" ""
 
-    # Trick configure system.
+    # Trick configure system
     sed -n 's,^ *path = \(.*\),\1,p' .gitmodules | while read path; do
```
Comment (IvarWithoutBones) [inline pkgs/misc/emulators/citra/default.nix:65]: This does the same thing as the `substituteInPlace` above, but if I recall correctly it's broken. Let's remove it.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Just a few comments 😊

**Comment by SuperSamus** [comment]
Done.
What about the remaining questions? (The overkill assert, whether `libusb` is devendored, and the breaking change)

**Comment by IvarWithoutBones** [comment]
Apologies about answering the questions a bit later than code review, I got busy the other day.

> I'm not sure if all of these options should be true by default. Should I respect Citra's CMakeLists.txt defaults or the more stuff is true the better?

I'd say the more stuff the better. Closure size isn't massively effected, so having CI build this with support for as much stuff as possible makes sense to me.

> I changed one option name. It's technically a breaking change, but I doubt that anyone overrode that option.

I think that should be fine, as you said it probably won't affect anyone.

> That assert maybe is overkill, if one simply has both options true the only effect is that FDK won't be enabled.

IMO can't hurt to have it, it can potentially save a fair bit of debugging.

> I didn't put any option for ENABLE_QT_TRANSLATION.

Why not?

> Is libusb being devendored if it's just being put in buildInputs?

Yes! See https://github.com/citra-emu/citra/blob/324e88d6e08ab8bc95671e6ba8011fc8812ac2c9/CMakeLists.txt#L205.

**Review by IvarWithoutBones — APPROVED** [review]
LGTM! I'm really happy seeing citra get some love. thanks for the work!

**Comment by SuperSamus** [comment]
I didn't put `ENABLE_QT_TRANSLATION` because I didn't know what dependency it needed.
I should have just taken a look at Yuzu.

**Comment by SuperSamus** [comment]
I found out that if I build without Qt, Citra won't find the GPU drivers:
```
libGL error: MESA-LOADER: failed to open radeonsi: /nix/store/kgzhnlza09sw2kzzvmp3jp0bypsglqkp-libdrm-2.4.108/lib/libdrm_nouveau.so.2: undefined symbol: drmCloseBufferHandle (search paths /run/opengl-driver/lib/dri, suffix _dri)
libGL error: failed to load driver: radeonsi
libGL error: MESA-LOADER: failed to open swrast: /nix/store/kgzhnlza09sw2kzzvmp3jp0bypsglqkp-libdrm-2.4.108/lib/libdrm_nouveau.so.2: undefined symbol: drmCloseBufferHandle (search paths /run/opengl-driver/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  149 (GLX)
  Minor opcode of failed request:  3 (X_GLXCreateContext)
  Value in failed request:  0x0
  Serial number of failed request:  100
  Current serial number in output stream:  101
```

I should have tested for this...
Why is it looking for `radeonsi` and not `amdgpu` though?

**Comment by SuperSamus** [comment]
I rebased, and the issue is gone.
No idea why.

**Review by abbradar — APPROVED** [review]

**Pull request merged**